File: Clash of Code/Find Brian/Find Brian.rb
H = gets.to_i
W = gets.to_i
b=nil
G=H.times.map{|y|
    r=gets.chomp.chars
    W.times{|x|
        b=[x,y]if r[x]==?B
    }
    r
}

A=[]
if b
    [[b[0]+1,b[1]],[b[0]-1,b[1]],[b[0],b[1]+1],[b[0],b[1]-1],].each{|x,y|
        A<<G[y][x] if x>=0&&x<W&&y>=0&&y<H&&G[y][x]!=?*
    }
end

L={
    "K" => "KITCHEN",
    "T" => "BATHROOM",
    "L" => "LIVING ROOM",
    "E" => "BEDROOM",
    "C" => "CORRIDOR",
}

puts A==[] ? "BRIAN IS NOWHERE" : "BRIAN IS IN THE #{L[A[0]]}"

File: Clash of Code/Find Brian/Find Brian.py
import sys
import math

h = int(input())
w = int(input())
lastline=""
for i in range(h):
    a = input()
    for j in range(len(a)):
        if a[j]=="B":
            x=""
            if j>0 and a[j-1]!="*":
                x=a[j-1]
            elif j+1<len(a) and a[j+1]!="*":
                x=a[j+1]
            elif i>0 and lastline[j]!="*":
                x=lastline[j]
            else:
                a=input()
                x=a[j]
                i+=1
            if x=="K":
                print("BRIAN IS IN THE KITCHEN")
            elif x=="C":
                print("BRIAN IS IN THE CORRIDOR")
            elif x=="T":
                print("BRIAN IS IN THE BATHROOM")
            elif x=="L":
                print("BRIAN IS IN THE LIVING ROOM")
            elif x=="E":
                print("BRIAN IS IN THE BEDROOM")
            elif x=="S":
                print("BRIAN IS IN THE SWIMMING POOL")
            else:
                print("BRIAN IS NOWHERE")  
    lastline=a

